File: java/timesheet/src/main/java/timesheet/Employee.java
package timesheet;

import java.time.LocalDate;
import java.time.LocalDateTime;

public class Employee implements Worker {
    private final String firstName;
    private final String lastName;
    private final LocalDate dateOfBirth;
    private final String email;

    private Timesheet timesheet;

    Employee(String firstName, String lastName, LocalDate dateOfBirth, String email) {
        this.firstName = firstName;
        this.lastName = lastName;
        this.dateOfBirth = dateOfBirth;
        this.email = email;
    }

    Employee(String firstName, String lastName, LocalDate dateOfBirth, String email, Timesheet timesheet) {
        this.firstName = firstName;
        this.lastName = lastName;
        this.dateOfBirth = dateOfBirth;
        this.email = email;
        this.timesheet = timesheet;
    }

    @Override
    public String getFirstName() {
        return firstName;
    }

    public String getLastName() {
        return lastName;
    }

    public LocalDate getDateOfBirth() {
        return dateOfBirth;
    }

    @Override
    public String getEmail() {
        return email;
    }

    @Override
    public void work() {
        //working...
    }

    @Override
    public Timesheet getTimesheet() {
        return timesheet;
    }
}

File: scala/timesheet/src/main/scala/timesheet/Employee.scala
package timesheet

import java.time.LocalDate

class Employee(val firstName: String, val lastName: String, val dateOfBirth: LocalDate, val email: String, var timesheet: Option[Timesheet] = None) extends Worker {

  def this(firstName: String, lastName: String, dateOfBirth: LocalDate, email: String) = {
    this(firstName, lastName, dateOfBirth, email, None)
  }

  override def getFirstName: String = firstName

  def getLastName: String = lastName

  def getDateOfBirth: LocalDate = dateOfBirth

  override def getEmail: String = email

  override def work(): Unit = {
    // working...
  }

  override def getTimesheet: Timesheet = timesheet.getOrElse(null)
}